RoFILib/snake_reconfig/Snake_algorithms.h

```diff
@@ -0,0 +1,280 @@
+    return sg.getDist();
+}
+
+inline std::vector<Configuration> SnakeStar(const Configuration& init, AlgorithmStat* stat = nullptr, 
+    int limit = 5000, double path_pref = 0.1)
+{
+    unsigned step = 90;
+    double free_pref = 1 - path_pref;
+    ConfigPred pred;
+    ConfigPool pool;
+
+    SpaceGrid grid(init.getIDs().size());
+    // Already computed shortest distance from init to configuration.
+    ConfigValue initDist;
+
+    // Shortest distance from init through this configuration to goal.
+    ConfigValue goalDist;
+
+    double startDist = eval(init, grid);
+
+    if (startDist == 0)
+        return {init};
+
+    MinMaxHeap<EvalPair, SnakeEvalCompare> queue(limit);
+
+    const Configuration* pointer = pool.insert(init);
+    const Configuration* bestConfig = pointer;
+    unsigned bestScore = startDist;
+    double worstDist = startDist;
+
+    initDist[pointer] = 0;
+    goalDist[pointer] = startDist;
+    pred[pointer] = pointer;
+    int maxQSize = 0;
+    int i = 0;
+    queue.push( {goalDist[pointer], pointer} );
+
+
+    while (!queue.empty() && i++ < limit) {
+        maxQSize = std::max(maxQSize, queue.size());
+        const auto [d, current] = queue.popMin();
+        double currDist = initDist[current];
+
+        std::vector<Configuration> nextCfgs;
+        simpleNext(*current, nextCfgs, step);
+
+        for (const auto& next : nextCfgs) {
+            const Configuration* pointerNext;
+            double newEval = eval(next, grid);
+            double newDist = path_pref * (currDist + 1) + free_pref * newEval;
+            bool update = false;
+
+            if (newEval != 0 && limit <= queue.size() + i) {
+                if (newDist > worstDist)
+                    continue;
+
+                worstDist = newDist;
+                if (!queue.empty())
+                    queue.popMax();
+            }
+
+            if (newDist > worstDist)
+                worstDist = newDist;
+
+            if (!pool.find(next)) {
+                pointerNext = pool.insert(next);
+                initDist[pointerNext] = currDist + 1;
+                update = true;
+            } else {
+                pointerNext = pool.get(next);
+            }
+
+            if (newEval < bestScore) {
+                bestScore = newEval;
+                bestConfig = pointerNext;
+            }
+
+            if ((currDist + 1 < initDist[pointerNext]) || update) {
+                initDist[pointerNext] = currDist + 1;
+                goalDist[pointerNext] = newDist;
+                pred[pointerNext] = current;
+                queue.push({newDist, pointerNext});
+            }
+
+            if (newEval == 0) {
+                auto path = createPath(pred, pointerNext);
+                if (stat != nullptr) {
+                    stat->pathLength = path.size();
+                    stat->queueSize = maxQSize;
+                    stat->seenCfgs = pool.size();
+                }
+                return path;
+            }
```

Comment:
The body is hard to follow. However, it is for a discussion on how to split it in pices.

Reply:
Could we leave this change for future work on my part of the code? (For new branch after merge)